File: internal/content/testdata/bench/typescript/nc-letter-combinations.ts
function letter_combinations(digits: string): string[] {
    if (!digits) return [];
    const mapping: {[key: string]: string} = {
        "2": "abc", "3": "def", "4": "ghi", "5": "jkl",
        "6": "mno", "7": "pqrs", "8": "tuv", "9": "wxyz"
    };
    const res: string[] = [];
    function backtrack(index: number, current: string): void {
        if (index === digits.length) {
            res.push(current);
            return;
        }
        for (const ch of mapping[digits[index]]) {
            backtrack(index + 1, current + ch);
        }
    }
    backtrack(0, "");
    res.sort();
    return res;
}

File: internal/content/testdata/bench/go/nc-letter-combinations.go
import "sort"

func letter_combinations(digits string) []string {
	if len(digits) == 0 {
		return []string{}
	}
	mapping := map[byte]string{
		'2': "abc", '3': "def", '4': "ghi", '5': "jkl",
		'6': "mno", '7': "pqrs", '8': "tuv", '9': "wxyz",
	}
	res := []string{}
	var backtrack func(index int, current string)
	backtrack = func(index int, current string) {
		if index == len(digits) {
			res = append(res, current)
			return
		}
		for _, ch := range mapping[digits[index]] {
			backtrack(index+1, current+string(ch))
		}
	}
	backtrack(0, "")
	sort.Strings(res)
	return res
}
